Show real host when the user has a hiddenhost (not just an account) or a sethost.

diff -r da4fffdd6fc1 ircd/m_check.c
--- a/ircd/m_check.c
+++ b/ircd/m_check.c
@@ -384,7 +384,7 @@
   ircd_ntoa(&(cli_ip(acptr))));
   send_reply(sptr, RPL_DATASTR, outbuf);
 
-  if (IsSetHost(acptr) || IsAccount(acptr)) {
+  if (HasHiddenHost(acptr) || IsSetHost(acptr)) {
     ircd_snprintf(0, outbuf, sizeof(outbuf), " Real User/Host:: %s@%s", cli_user(acptr)->realusername, cli_user(acptr)->realhost);
     send_reply(sptr, RPL_DATASTR, outbuf);
   }
